app/controllers/api/discussions_controller.rb

```diff
@@ -3,60 +3,44 @@ class API::DiscussionsController < API::RestfulController
   load_resource only: [:create, :update]
 
   def dashboard
-    instantiate_collection { |collection| filter_collection collection }
-    respond_with_discussions
+    instantiate_collection { |collection| collection_for_dashboard collection }
+    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
+  end
+
+  def inbox
+    instantiate_collection { |collection| collection_for_inbox collection }
+    respond_with_collection scope: { visible_in_inbox: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
   end
 
   def index
     load_and_authorize :group if params[:group_id] || params[:group_key]
     instantiate_collection
-    respond_with_discussions
-  end
-
-  def show
-    respond_with_discussion
+    respond_with_collection serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
   end
 
   private
 
-  def respond_with_discussion
-    if resource.errors.empty?
-      render json: DiscussionWrapper.new(discussion: resource, discussion_reader: discussion_reader),
-             serializer: DiscussionWrapperSerializer,
-             root: 'discussion_wrappers'
-    else
-      respond_with_errors
-    end
-  end
-
-  def respond_with_discussions
-    render json: DiscussionWrapper.new_collection(user: current_user, discussions: @discussions),
-           each_serializer: DiscussionWrapperSerializer,
-           root: 'discussion_wrappers'
-  end
-
   def visible_records
-    Queries::VisibleDiscussions.new(user: current_user, groups: visible_groups).sorted_by_latest_activity
+    Queries::VisibleDiscussions.new(user: current_user, groups: visible_groups)
   end
 
-  private
-
   def visible_groups
     Array(@group).presence || current_user.groups
   end
 
-  def filter_collection(collection, filter = params[:filter])
+  def collection_for_dashboard(collection, filter: params[:filter])
     case filter
-    when 'show_proposals'     then collection.not_muted.with_active_motions
-    when 'show_participating' then collection.not_muted.participating
-    when 'show_starred'       then collection.not_muted.starred
-    when 'show_muted'         then collection.muted
-    when 'show_unread'        then collection.not_muted.unread
-    else                           collection.not_muted
+    when 'show_participating' then collection.not_muted.participating.sorted_by_importance
+    when 'show_muted'         then collection.muted.sorted_by_latest_activity
+    else                           collection.not_muted.sorted_by_importance
     end
   end
 
-  def discussion_reader
-    @dr ||= DiscussionReader.for(user: current_user, discussion: @discussion)
+  def collection_for_inbox(collection)
+    collection.not_muted.unread.sorted_by_latest_activity
+  end
+
+  def collection=(value)
```

Comment:
This is overriding the collection= method on RestfulController

On Mon, Aug 3, 2015 at 2:05 AM, Robert Guthrie <notifications@github.com>
wrote:

> In app/controllers/api/discussions_controller.rb
> <https://github.com/loomio/loomio/pull/2369#discussion_r36067221>:
>
> >      end
> >    end
> >
> > -  def discussion_reader
> > -    @dr ||= DiscussionReader.for(user: current_user, discussion: @discussion)
> > +  def collection_for_inbox(collection)
> > +    collection.not_muted.unread.sorted_by_latest_activity
> > +  end
> > +
> > +  def collection=(value)
>
> I would like to suggest that you change the name of this method.. I think
> the syntatic sugar his hiding that there is work going on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r36067221>.
>